test/agent-policy-tests/test-script-utils.bats

```diff
@@ -0,0 +1,283 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+####################################################################
+## This script contains unit tests for
+## modules/agent-policy/scripts/script-utils.sh
+## This script is ran via `make docker_test_bats`
+####################################################################
+
+PROJECT_ID="test-project-id"
+POLICY_ID="ops-agents-test-policy"
+DESCRIPTION="an example test policy"
+AGENT_RULES_JSON_BASIC="[{\"type\":\"metrics\"}]"
+OS_TYPES_JSON_BASIC="[{\"short_name\":\"centos\",\"version\":\"8\"}]"
+EMPTY_LIST_JSON="[]"
+ETAG="db5c5a61-6a05-4f67-8f84-c89a654eb576"
+
+load "/usr/local/bats-support/load.bash"
+load "/usr/local/bats-assert/load.bash"
+
+
+# Setup before every test
+setup() {
+    SCRIPT_DIR="${BATS_TEST_DIRNAME%/}"
+    UTILS_ABS_PATH="${SCRIPT_DIR}/../../modules/agent-policy/scripts/script-utils.sh"
+    source "$UTILS_ABS_PATH"
+}
+
+
+##############################################################
+##  get_create_command tests                                ##
+##############################################################
+
+@test "Tests get_create_command with defaults" {
+    local description=""
+    local agent_rules_json="$AGENT_RULES_JSON_BASIC"
+    local group_labels_json="$EMPTY_LIST_JSON"
+    local os_types_json="$OS_TYPES_JSON_BASIC"
+    local zones_json="$EMPTY_LIST_JSON"
+    local instances_json="$EMPTY_LIST_JSON"
+
+    local expected_command="gcloud alpha compute instances ops-agents"
+    expected_command="$expected_command policies create ops-agents-test-policy"
+    expected_command="$expected_command --agent-rules='type=metrics'"
+    expected_command="$expected_command --os-types='version=8,short-name=centos'"
+    expected_command="$expected_command --project='test-project-id' --quiet"
+
+    run get_create_command "$PROJECT_ID" "$POLICY_ID" \
+        "$description" "$agent_rules_json" "$group_labels_json" "$os_types_json" \
+        "$zones_json" "$instances_json"
+
+    assert_equal "$output" "$expected_command"
+}
+
+@test "Tests get_create_command with description" {
+    local description="$DESCRIPTION"
```

Comment:
Do we have to mirror these values locally? I', not sure why we wouldn't just reference them directly.

Reply:
Got it.